emulator_test.go

```diff
@@ -214,36 +222,35 @@ func TestSuccessTaskExecution(t *testing.T) {
 		},
 	}
 	createdTask, err := client.CreateTask(context.Background(), &createTaskRequest)
+	require.NoError(t, err)
 
 	getTaskRequest := taskspb.GetTaskRequest{
 		Name: createdTask.GetName(),
 	}
 
-	// Need to give it a chance to make the actual call
-	time.Sleep(100 * time.Millisecond)
+	receivedRequest, err := awaitHttpRequest(receivedRequests)
+	require.NoError(t, err)
+
 	gettedTask, err := client.GetTask(context.Background(), &getTaskRequest)
 	assert.Error(t, err)
 	assert.Nil(t, gettedTask)
 
 	// Validate that the call was actually made properly
-	assert.NotNil(t, receivedRequest, "Request was received")
-
-	// Simple predictable headers
-	expectHeaders := map[string]string{
-		"X-CloudTasks-TaskExecutionCount": "0",
-		"X-CloudTasks-TaskRetryCount":     "0",
-		"X-CloudTasks-TaskName":           "my-test-task",
-		"X-CloudTasks-QueueName":          "test",
-	}
-	actualHeaders := make(map[string]string)
-	for hdr := range expectHeaders {
-		actualHeaders[hdr] = receivedRequest.Header.Get(hdr)
+	if assert.NotNil(t, receivedRequest, "Request was received") {
```

Comment:
I'm not a fan of this - can you replace this with `require.NotNil` to avoid the scenario you were mentioning instead?

Reply:
Sorry, should have gone back to that - I'd not planned such a big change when I tackled the panics and previously there was teardown code that ran after these. Since I've now refactored those to `defer xxx` earlier in the func then I agree `require.XX` is much cleaner. Done now.